src/goGraph/uiPatchHandler.ts

```diff
@@ -0,0 +1,257 @@
+import go from 'gojs';
+import {
+	IUiPatchHandler,
+	UiConnectorPatchProperties,
+	UiEdge,
+	UiEdgePatchProperties,
+	UiNodeConnector,
+	UiNodePatchProperties,
+	UiNodeSymbol,
+} from '../core/ui/uiNegotiator';
+import { getNodeSymbolTemplate } from '../symbol-api';
+import { BaseNodeData, EdgeData, NodeUiCategory, NodeUiItemCategory, PortData } from './types';
+
+const nodePropMap: Record<keyof UiNodePatchProperties, string> = {
+	backgroundColor: 'fill',
+	borderColor: 'stroke',
+	highlight: 'highlight',
+	highlightBorderColor: 'highlightBorderColor',
+	label: 'label',
+	shape: 'shape',
+	symbolId: 'symbolId',
+	nodeTemplate: 'nodeTemplate',
+	symbolHeight: 'symbolHeight',
+	symbolWidth: 'symbolWidth',
+};
+
+const edgePropMap: Record<keyof UiEdgePatchProperties, string> = {
+	color: 'color',
+	highlight: 'highlight',
+	highlightColor: 'highlightColor',
+	thickness: 'stroke',
+};
+
+export class GoJsPatchHandler implements IUiPatchHandler {
```

Comment:
This is class is awesome. Everything that is difficult here is related to gojs - api. All complex graph stuff is abstracted away before we get to this point. I think introducing IUiPatchHandler interface and UiNegotiator to divide responsibility was a great idea 👍 👍

Reply:
We keep this:)